src/reports/reportData.ts

```diff
@@ -0,0 +1,85 @@
+import * as systeminformation from "systeminformation";
+
+const UNKNOWN_SPEED = -1;
+const UNKNOWN_CORES_AMOUNT = -1;
+const UNKNOWN_MEMORY_SIZE = -1;
+
+const byteToGigabyte = (bytes: number): number => bytes / 1e9;
+
+export type OsInfo = {
+  platform: string;
+  distro: string;
+  arch: string;
+  kernel: string;
+};
+
+export type CpuInfo = {
+  manufacturer: string;
+  brand: string;
+  speedGHz: number;
+  cores: number;
+};
+
+export type Specs = {
+  os: OsInfo;
+  cpu: CpuInfo;
+  memoryGB: number;
+};
+
+export type ReportData = {
+  timestamp: string;
+  os: string;
+  kernel: string;
+  cpu: string;
+  cores: string;
+  speedGHz: string;
+  memoryGB: string;
+};
+
+let specsCache: Promise<Specs> | undefined;
+
+async function getSpecs(): Promise<Specs> {
+  const cpuData = await systeminformation.cpu();
```

Comment:
use Promise.all()

Reply:
why, its more readable this way

Reply:
it is not the same... `Promise.all` is "concurrent" (as any single request does not block the execution of that method, only once all requests happened then execution is blocked)

Reply:
oh i see, makes sense. will change

Reply:
by calling them this way(your, original ) you are actually  executing them in serial(the second call would not be called until first is resolved) it. is ok if  each call is dependent on the previous , now i'm not sure what `systeminformation` is doing under the hood and why the API is async, but if it is using some IO, then using Promise.all will handle these promises in parallel.